modules/homebrew.nix

```
@@ -1,219 +1,751 @@
+          ${mkDocOptionLink "environment.variables"}.
+        '';
+      };
+      noLock = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to disable lockfile generation when you manually invoke
+          <command>brew bundle [install]</command>. This is often desirable when
+          ${mkDocOptionLink "homebrew.global.brewfile"} is enabled, since
+          <command>brew bundle [install]</command> will try to write the lockfile in the Nix store,
+          and complain that it can't (though the command will run successfully regardless).
+
+          Implementation note: when enabled, this option sets the
+          <literal>HOMEBREW_BUNDLE_NO_LOCK</literal> environment variable, by adding it to
+          ${mkDocOptionLink "environment.variables"}.
+        '';
+      };
+
+      homebrewEnvironmentVariables = mkInternalOption { type = types.attrs; };
+    };
+
+    config = {
+      homebrewEnvironmentVariables = {
+        HOMEBREW_BUNDLE_FILE = mkIf config.brewfile "${brewfileFile}";
+        HOMEBREW_BUNDLE_NO_LOCK = mkIf config.noLock "1";
+        HOMEBREW_NO_AUTO_UPDATE = mkIf config.noAutoUpdate "1";
+      };
+    };
+  };
+
+
+  tapOptions = { config, ... }: {
+    options = {
+      name = mkOption {
+        type = types.str;
+        example = "homebrew/cask-fonts";
+        description = ''
+          When <option>clone_target</option> is unspecified, this is the name of a formula
+          repository to tap from GitHub using HTTPS. For example, <literal>"user/repo"</literal>
+          will tap https://github.com/user/homebrew-repo.
+        '';
+      };
+      clone_target = mkNullOrStrOption {
+        description = ''
+          Use this option to tap a formula repository from anywhere, using any transport protocol
+          that <command>git</command> handles. When <option>clone_target</option> is specified, taps
+          can be cloned from places other than GitHub and using protocols other than HTTPS, e.g.,
+          SSH, git, HTTP, FTP(S), rsync.
+        '';
+      };
+      force_auto_update = mkNullOrBoolOption {
+        description = ''
+          Whether to auto-update the tap even if it is not hosted on GitHub. By default, only taps
+          hosted on GitHub are auto-updated (for performance reasons).
+        '';
+      };
+
+      brewfileLine = mkInternalOption { type = types.nullOr types.str; };
+    };
+
+    config =
+      let
+        sCfg = mkProcessedSubmodConfig config;
+      in
+      {
+        brewfileLine =
+          "tap ${sCfg.name}"
+          + optionalString (sCfg ? clone_target) ", ${sCfg.clone_target}"
+          + optionalString (sCfg ? force_auto_update)
+            ", force_auto_update: ${sCfg.force_auto_update}";
+      };
+  };
+
+  # Sourced from https://docs.brew.sh/Manpage#global-cask-options
+  # and valid value for `HOMEBREW_CASK_OPTS`.
```

```suggestion
  # and valid values for `HOMEBREW_CASK_OPTS`.
```

At least this reads as if it should be plural.

---

Yup, thanks for catching this.